### PR TITLE
convert example4.py into example4.cpp #15651

### PR DESCRIPTION
**Pull request by mohitsatr**
Resolves: #15555

### CODE REVIEW

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example4.cpp [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+    <property name="fileExtensions" value="java, xml, cpp"/>
   </module>
 </module>
 */
 // violation 7 lines above 'File does not end with a newline.'
 // xdoc section -- start
-print("Hello world!")
-# no ⤶ below it is violation
+void main() { // ⤶
+    std::cout << "Hello World!";// ⤶
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example4.cpp:11]: This caused spelling violation.
Can we have some cpp code that is written in full English words?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example4.cpp (hunk 1 above)
Comment (mohitsatr) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example4.cpp:11]: sure we can either 
```suggestion
    char s[] = "Hello World!";// ⤶
```
or simply 
```suggestion
    return 0;// ⤶
```

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/Example4.cpp:11]: Please do first

**Review by romani — APPROVED** [review]
Ok to merge if ci pas

**Comment by romani** [comment]
@mohitsatr , thanks a lot for update!!!

**Comment by mohitsatr** [comment]
@romani thanks for guiding me through the process. looking forward to learn more of checkstyle's internals and make meaningful contributions.

**Pull request merged**